### PR TITLE
fix: variableBorrows needs more 0.01% when repay

### PR DESCRIPTION
**Pull request by SteveYuOWO**

### CODE REVIEW

File: src/AAVE/Modal/WithdrawModal.jsx [hunk 1]
```diff
@@ -268,17 +272,34 @@ function update() {
...
+      : Big(availableLiquidityAmount)
+          .mul(Big(10).pow(decimals))
+          .toFixed(0, ROUND_DOWN)
+    : "0";
+const shownMaxValue =
+  isValid(underlyingBalance) && isValid(availableLiquidityAmount)
+    ? bigMin(underlyingBalance, availableLiquidityAmount).toFixed()
+    : "0";
+
+console.log({
```
Comment (think-in-universe) [inline src/AAVE/Modal/WithdrawModal.jsx:294]: remove logging

File: src/AAVE/Modal/WithdrawModal.jsx [hunk 2]
```diff
@@ -436,6 +459,11 @@ return (
                 config,
                 loading: state.loading,
                 children: `Approve ${symbol}`,
+                disabled:
+                  !isValid(state.newHealthFactor) ||
+                  state.newHealthFactor === "" ||
+                  (Big(state.newHealthFactor).lt(1) &&
```
Comment (think-in-universe) [inline src/AAVE/Modal/WithdrawModal.jsx:465]: Why disable button if `health factor < 1`?

File: src/AAVE/Modal/WithdrawModal.jsx [hunk 3]
```diff
@@ -464,16 +492,38 @@ return (
...
                 onClick: () => {
-                  const amount = Big(state.amount)
+                  const actualAmount =
+                    state.amount === shownMaxValue
+                      ? actualMaxValue
+                      : Big(state.amount)
+                          .mul(Big(10).pow(decimals))
+                          .toFixed(0, ROUND_DOWN);
+                  const shownAmount = Big(
+                    state.amount === shownMaxValue
```
Comment (think-in-universe) [inline src/AAVE/Modal/WithdrawModal.jsx:508]: Why not use `state.amount` directly in LINE 508~510?

File: src/AAVE/Modal/WithdrawModal.jsx [hunk 4]
```diff
@@ -464,16 +492,38 @@ return (
...
+                      ? shownMaxValue
+                      : state.amount
+                  )
                     .mul(Big(10).pow(decimals))
                     .toFixed(0);
+                  console.log({
```
Comment (think-in-universe) [inline src/AAVE/Modal/WithdrawModal.jsx:514]: remove logging?

File: src/AAVE/Modal/RepayModal.jsx [hunk 5]
```diff
@@ -111,7 +119,36 @@ State.init({
...
 });
 
-const maxValue = variableBorrows;
+function bigMin(_a, _b) {
+  const a = Big(_a);
+  const b = Big(_b);
+  return a.gt(b) ? b : a;
+}
+
+function getBalanceForCalculate() {
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:128]:
```suggestion
function getAvailableBalance() {
```

File: src/AAVE/Modal/RepayModal.jsx [hunk 6]
```diff
@@ -501,13 +545,24 @@ return (
...
-                  const amount = Big(state.amount)
+                  const actualAmount = Big(
+                    state.amount === shownMaxValue
+                      ? actualMaxValue
+                      : state.amount
+                  )
+                    .mul(Big(10).pow(decimals))
+                    .toFixed(0);
+                  const shownAmount = Big(
+                    state.amount === shownMaxValue
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:556]: Why not use `state.amount` directly in LINE 556~558?

**Review by think-in-universe — CHANGES_REQUESTED** [review]

File: src/AAVE/Modal/WithdrawModal.jsx (hunk 2 above)
Comment (SteveYuOWO) [reply src/AAVE/Modal/WithdrawModal.jsx:459]: When `health factor < 1`, the user has borrowed money. it is easy to be liquidated

File: src/AAVE/Modal/RepayModal.jsx [hunk 7]
```diff
@@ -310,21 +348,19 @@ function repayETH(amount) {
...
         )
         .then((tx) => {
           tx.wait().then((res) => {
             const { status } = res;
             if (status === 1) {
               onActionSuccess({
-                msg: `You repaied ${Big(amount)
-                  .div(Big(10).pow(decimals))
-                  .toFixed(8)} ${symbol}`,
+                msg: `You repaied ${Big(shownAmount).toFixed(8)} ${symbol}`,
```
Comment (think-in-universe) [inline src/AAVE/Modal/RepayModal.jsx:363]:
```suggestion
                msg: `You repaid ${Big(shownAmount).toFixed(8)} ${symbol}`,
```

**Review by think-in-universe — CHANGES_REQUESTED** [review]

File: src/AAVE/Modal/WithdrawModal.jsx (hunk 2 above)
Comment (think-in-universe) [reply src/AAVE/Modal/WithdrawModal.jsx:455]: oh, yeah. that's right.

**Pull request merged**